### DISCUSSION TITLE
'error sending data frame: Too many frames in the outbound queue ' transfer of large size data fails

### DISCUSSION THREAD

**Discussion by babydeveloper99** [istio/istio#54235] [General]
Issue is I have a service which sends data of size almost 1GB I know this is not best practice but what we see is with Istio-proxy over http/1.1 there is no issue but when I switch the protocol to http/2 I see the request fails with below log
external/envoy/source/common/http/http2/codec_impl.cc:677  [Tags: "ConnectionId":"51000"] error sending data frame: Too many frames in the outbound queue thread=31
[2024-12-03T12:05:05.334Z] "POST /upload HTTP/1.1" 502 UPE upstream_reset_before_response_started{protocol_error} - "-" 685768499 87 3653 - "-" "Go-http-client/1.1" "442a923e-8742-4508-93ff-a76f5a5906c3" "consumer-app-service.shiva.svc.cluster.local:8080" "10.42.0.138:8080" outbound|8080||consumer-app-service.shiva.svc.cluster.local
Can someone please explain why I am getting this issue ? and suggest if there is any solution ?

**Comment by zirain** [comment]
https://github.com/istio/istio/issues/40617 help?

**Reply by babydeveloper99** [reply]
Hi @zirain  thanks for your reply I have already checked #40617 it didn't help as it suggest to increase max_outbound_frames. I guess this issue is with Flow control but need a confirmation if it is ? what can be done?
